client/x/evmstaking/keeper/withdrawal_queue_test.go

```diff
@@ -0,0 +1,246 @@
+			expectedLen: len(withdrawals),
+			expected:    withdrawals,
+		},
+		{
+			name:        "Dequeue with empty queue",
+			maxDequeue:  3,
+			expectedLen: 0,
+			expected:    []types.Withdrawal{},
+		},
+	}
+
+	for _, tc := range tcs {
+		s.Run(tc.name, func() {
+			s.initQueue()
+
+			if tc.name != "Dequeue with empty queue" {
+				s.addWithdrawals(withdrawals)
+			}
+
+			// Set max dequeue parameter
+			params, err := s.EVMStakingKeeper.GetParams(s.Ctx)
+			s.NoError(err)
+			params.MaxWithdrawalPerBlock = tc.maxDequeue
+			err = s.EVMStakingKeeper.SetParams(s.Ctx, params)
+			s.NoError(err)
+
+			queueLen := s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx)
+
+			// Dequeue the withdrawals
+			result, err := s.EVMStakingKeeper.DequeueEligibleWithdrawals(s.Ctx)
+			s.NoError(err)
+			s.Equal(tc.expectedLen, len(result))
+
+			// Check Queue length is decreased by the number of dequeued withdrawals
+			s.Equal(
+				queueLen-uint64(tc.expectedLen),
+				s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx),
+			)
+
+			// Validate the content of the dequeued withdrawals
+			for i, w := range result {
+				s.Equal(tc.expected[i].ExecutionAddress, w.Address.String())
+				s.Equal(tc.expected[i].Amount, w.Amount)
+			}
+		})
+	}
+}
+
+func (s *TestSuite) TestPeekEligibleWithdrawals() {
+	tcs := []struct {
+		name        string
+		maxDequeue  uint32
+		expectedLen int
+		expected    []types.Withdrawal
+	}{
+		{
+			name:        "Peek 1 withdrawal",
+			maxDequeue:  1,
+			expectedLen: 1,
+			expected:    withdrawals[:1],
+		},
+		{
+			name:        "Peek 2 withdrawals",
+			maxDequeue:  2,
+			expectedLen: 2,
+			expected:    withdrawals[:2],
+		},
+		{
+			name:        "Peek more than available",
+			maxDequeue:  10,
+			expectedLen: len(withdrawals),
+			expected:    withdrawals,
+		},
+		{
+			name:        "Peek with empty queue",
+			maxDequeue:  3,
+			expectedLen: 0,
+			expected:    []types.Withdrawal{},
+		},
+	}
+
+	for _, tc := range tcs {
+		s.Run(tc.name, func() {
+			s.initQueue()
+			if tc.name != "Peek with empty queue" {
+				s.addWithdrawals(withdrawals)
+			}
+
+			// Set max dequeue parameter
+			params, err := s.EVMStakingKeeper.GetParams(s.Ctx)
+			s.NoError(err)
+			params.MaxWithdrawalPerBlock = tc.maxDequeue
+			err = s.EVMStakingKeeper.SetParams(s.Ctx, params)
+			s.NoError(err)
+
+			queueLen := s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx)
+
+			// Peek the withdrawals
+			result, err := s.EVMStakingKeeper.PeekEligibleWithdrawals(s.Ctx)
+			s.NoError(err)
+			s.Equal(tc.expectedLen, len(result))
+
+			// Peek does not change the queue length
+			s.Equal(queueLen, s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx))
+
+			// Validate the content of the dequeued withdrawals
+			for i, w := range result {
+				s.Equal(tc.expected[i].ExecutionAddress, w.Address.String())
+				s.Equal(tc.expected[i].Amount, w.Amount)
+			}
+		})
+	}
+}
+
+func (s *TestSuite) TestGetAllWithdrawals() {
+	s.initQueue()
+
+	// Add a withdrawal to the queue
```

Comment:
Add a test to withdraw for an empty queue

Reply:
@limengformal 
Applied at https://github.com/piplabs/story/pull/61/commits/fdb9ab6306c7bc1382a0b12012d3df1ee4b78737